python/pyspark/ml/util.py

```diff
@@ -52,3 +71,141 @@ def _randomUID(cls):
         concatenates the class name, "_", and 12 random hex chars.
         """
         return cls.__name__ + "_" + uuid.uuid4().hex[12:]
+
+
+@inherit_doc
+class MLWriter(object):
+    """
+    Abstract class for utility classes that can save ML instances.
+
+    .. versionadded:: 2.0.0
+    """
+
+    def __init__(self, instance):
+        self._jwrite = instance._java_obj.write()
+
+    @since("2.0.0")
+    def save(self, path):
```

Comment:
Here we call the peer java object to do save operation, and the Params has been copied at the construction of the peer java object.